js/ascendex.js

```diff
@@ -467,55 +483,101 @@ module.exports = class ascendex extends Exchange {
         for (let i = 0; i < ids.length; i++) {
             const id = ids[i];
             const market = dataById[id];
-            const baseId = this.safeString (market, 'baseAsset');
-            const quoteId = this.safeString (market, 'quoteAsset');
-            const base = this.safeCurrencyCode (baseId);
-            const quote = this.safeCurrencyCode (quoteId);
+            let baseId = this.safeString (market, 'baseAsset');
+            let quoteId = this.safeString (market, 'quoteAsset');
+            const settleId = this.safeValue (market, 'settlementAsset');
+            let base = this.safeCurrencyCode (baseId);
+            let quote = this.safeCurrencyCode (quoteId);
+            const settle = this.safeCurrencyCode (settleId);
             const precision = {
                 'amount': this.safeNumber (market, 'lotSize'),
                 'price': this.safeNumber (market, 'tickSize'),
             };
             const status = this.safeString (market, 'status');
             const active = (status === 'Normal');
-            const type = ('useLot' in market) ? 'spot' : 'future';
+            const type = ('useLot' in market) ? 'spot' : 'swap';
             const spot = (type === 'spot');
-            const future = (type === 'future');
+            const perpetual = (type === 'swap');
             const margin = this.safeValue (market, 'marginTradable', false);
+            let contract = false;
+            if (margin || perpetual === true) {
+                contract = true;
+            }
+            let linear = false;
```

Comment:
Could just be `const linear = contract` assuming that all contracts on ascendex are linear